core/src/test/java/com/linecorp/armeria/internal/server/annotation/AnnotatedServiceResponseConverterTest.java

```diff
@@ -332,6 +344,31 @@ public Object expectOk() {
                                                  .of(HttpHeaderNames.of("x-custom-trailers"), "value"));
                 }
 
+                @Get("/http-file/expect-custom-header")
+                @AdditionalHeader(name = "x-custom-annotated-header", value = "annotated-value")
+                public HttpResult<HttpFile> httpFileExpectCustomHeader() {
+                    return HttpResult.of(HttpHeaders.of(HttpHeaderNames.of("x-custom-header"), "value"),
+                                         HTTPFILE);
+                }
+
+                @Get("/http-file/expect-custom-trailers")
+                @AdditionalTrailer(name = "x-custom-annotated-trailers", value = "annotated-value")
+                public HttpResult<HttpFile> httpFileExpectCustomTrailers() {
+                    return HttpResult.of(HttpHeaders.of(HttpHeaderNames.of("x-custom-header"), "value"),
+                                         HTTPFILE,
+                                         HttpHeaders.of(HttpHeaderNames.of("x-custom-trailers"), "value"));
+                }
```

Comment:
What is the expected result if a header is defined in both an `HttpResult` and an `AdditionalTrailer`?

```java
@Get("/http-file/expect-custom-trailers")
@AdditionalTrailer(name = "x-custom-annotated-trailers", value = "annotated-value")
@AdditionalTrailer(name = "x-custom-trailers", value = "default-value")
public HttpResult<HttpFile> httpFileExpectCustomTrailers() {
    return HttpResult.of(HttpHeaders.of(HttpHeaderNames.of("x-custom-header"), "value"),
                         HTTPFILE,
                         // Should override the default?
                         HttpHeaders.of(HttpHeaderNames.of("x-custom-trailers"), "value"));
}
```

Reply:
Trailers with the same name both in `@AdditionalTrailer` and `HttpResult`?
Then, the annotated one will overwrite the one in the `HttpResult`.

The same behavior with the other return types.

Reply:
The `@StatusCode(...)` will be overridden by `HttpResult`. Don't need the same approach for `AdditionalTrailer` and `AdditionalHeader`?

Reply:
But this behavior is common for other return types too.

Headers designated with annotations go into `ctx.additionalHeaders` and the other into `HttpResponse` and
it seems [this line](https://github.com/line/armeria/blob/20f72064565b8e0ca54aa8138822c57f1ab3f2b2/core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java#L169) is the cause of of this behavior.

`HttpHeadersUtil#mergeResponseHeaders` allows annotated headers(`ctx.additionalHeaders`) to overwrite the one published by `HttpResponse`.

Reply:
Changing the behavior of `Additional{Header,Trailer}` is a breaking change, so I guess we can't change that.

Reply:
Filed an issue for Armeria 2.x 😀
https://github.com/line/armeria/issues/3303